src/components/marker-chart/index.js

```diff
@@ -80,35 +87,36 @@ class MarkerChart extends React.PureComponent<Props> {
         </div>
         <MarkerChartCanvas
           key={threadIndex}
-          // ChartViewport props
-          isRowExpanded={isRowExpanded}
-          isSelected={isSelected}
-          timeRange={timeRange}
-          maxViewportHeight={maxViewportHeight}
-          maximumZoom={this.getMaximumZoom()}
-          selection={selection}
-          updateProfileSelection={updateProfileSelection}
-          viewportNeedsUpdate={viewportNeedsUpdate}
-          // MarkerChartCanvas props
-          rangeStart={timeRange.start}
-          rangeEnd={timeRange.end}
-          markerTimingRows={markerTimingRows}
-          maxMarkerRows={maxMarkerRows}
-          markers={markers}
-          rowHeight={ROW_HEIGHT}
+          viewportProps={{
```

Comment:
Breaking these out into objects greatly simplified the typing and made the errors MUCH easier to track down, and error messages MUCH simpler.